### DISCUSSION TITLE
why /_next/data/[hash]/xxx.json 404

### DISCUSSION THREAD

**Discussion by haifn** [vercel/next.js#38414] [Help]
I found some bad network requests in the logs，some like: /_next/data/[hash]/xxx.json
But I did not find the data directory in the .next directory.
how do i fix it ?

**Comment by icyJoseph** [comment]
This is most likely from a page that implements `getServerSideProps`, for example: ` http://localhost:3001/_next/data/UBmgmRU6qiWkZ9_9s40R7/tomato.json?hash=tomato` which I just made up, forcing it to always return `{ notFound: 404}`

How to fix this... well you'll have to figure out if there's a page that's catching stuff it shouldn't, or if it perhaps works as expected.

For example a root level catch all page, would intercept favicon requests, if no favicons exist in the `public` root.

**Comment by hafizasadabbas** [comment]
Facing the same problem not on local but on Vercel. Getting 404 in the network tab with the request URL just like
_next/data/build-id/slug.json?slug=slug

**Reply by shotor** [reply]
I'm getting this too, seemingly randomly the nextjs instance running in prod will start doing this after a while. And also setting the cache header to: 

```
cache-control: s-maxage=31536000, stale-while-revalidate
```

We are not returning a 404 at all. It's a static page and for all our 404s, we set revalidate to 1

Going to try manual revalidates. But so far the only thing that helps is restarting the instance.

Cannot find a way to reproduce either

**Comment by voronianski** [comment]
also getting this request - Next v13.0.5, still using `getInitialProps` method though

**Comment by RossMcMillan92** [comment]
We are getting this too. Similarly to [this article](https://arihantverma.com/posts/2021/07/18/workaround-nextjs-data-urls-same-domain-multiple-apps/), we run our next apps paths like example.com/app-1 and example.com/app-2. We use assetPath, and various other custom fixes to allow thism, but the calls to getServerSideProps don't utilise the assetPath and I can't figure out a way to fix it without relying on an nginx rewrite rule which is less than ideal.

The nginx rule, for those it may help:
``` nginx
location /_next/data/ {
    rewrite ^/_next/data/(.*)$ /app-1/_next/data/$1 redirect;
}
```

**Comment by WisaniShilumani** [comment]
Has anyone managed to find a work around for this without having to write nginx rules?

**Reply by voronianski** [reply]
the only thing that helped for now is to intercept these requests on the client before they are sent by re-writing `window.fetch` function -

```js
if (process.browser) {
  // intercept and prevent wasteful /_next/data/*.json request until Next.js issue is resolved
  // https://github.com/vercel/next.js/discussions/38414
  // https://github.com/vercel/next.js/issues/40611
  const { fetch: originalFetch } = window;
  const nextDataRequestRegex = /^\/_next\/data\/.*\.json/;
  window.fetch = async (...args) => {
    const [url] = args;
    if (nextDataRequestRegex.test(url)) {
      return Promise.reject();
    }
    return originalFetch(...args);
  };
}
```

**Reply by WisaniShilumani** [reply]
Thanks @voronianski , its not the best, but will have to do for now. Hopefully this can get fixed soon. Makes next.js feel less production ready

**Reply by gabor-at-reed** [reply]
Where do you have this hack in your app, @voronianski ?

**Reply by HovhannesMkoyan** [reply]
@voronianski can you please tell where should this snippet go in the app? where's the best place?

**Reply by jleslie02** [reply]
for those asking you can put this ina useEffect in the page that is causing the 404

**Reply by jleslie02** [reply]
Also thanks OP! this helped with microfrontends

**Reply by persocon** [reply]
this solution is great if you don't use `router.push` and etc :( with `router.push` it gives application error to you :'( sad would be perfect if it didn't

**Comment by moaazbhnas228** [comment]
Any updates on this?

**Comment by gabor-at-reed** [comment]
We have the same issue. I've just raised this as a bug here (with a proposed solution): https://github.com/vercel/next.js/issues/49926

I hope the Vercel guys will give us a solution soon! 🤞

**Comment by moaazbhnas228** [comment]
@WisaniShilumani 
@voronianski 
@haifn 
@RossMcMillan92
@gabor-at-reed 
Find a great explanation and solution for this issue here: https://raphael-leger.medium.com/react-webpack-chunkloaderror-loading-chunk-x-failed-ac385bd110e0

**Reply by gabor-at-reed** [reply]
it's not the same issue, I guess

**Comment by RossMcMillan92** [comment]
Not the same problem for me, but thanks for sharing!

**Comment by persocon** [comment]
using next 14 and having this issue now 🤷‍♀️

**Comment by igorfelluga** [comment]
I got same error but I found that when you use <Link> and the url linked does not exists your get this error
404 /_next/data/[hash]/xxx.json
Try to look for a link to  /xxx

**Comment by yubinjodev** [comment]
I'm getting a 403 on my end, does anyone know what this means? Our project does not even use getServerSideProps or getInitialProps

**Reply by alex-lechner** [reply]
Replace every `<Link href="" ...>...</Link>` with good ol' HTML links `<a href="" ...>...</a>` and the errors should disappear. At least that was the case for my project.

**Comment by sunjones0404** [comment]
Interim solution I've implemented is to add this check inside of getStaticProps on the page that triggers the 404

```
// Update your condition with your given dynamic path segment e.g. slug, hash, id etc.

if (slug === '[...slug]') {
      return {
        redirect: {
          destination: '/404',
          permanent: false
        }
      };
    }
```
